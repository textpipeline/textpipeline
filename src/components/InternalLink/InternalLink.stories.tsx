import { storiesOf } from '@storybook/react';
import React from 'react';

import InternalLink from './InternalLink';

storiesOf('InternalLink', module).add('default', () => <InternalLink to="/">Test Link</InternalLink>);
