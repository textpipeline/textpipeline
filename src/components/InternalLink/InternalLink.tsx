import Link from '@material-ui/core/Link';
import React from 'react';
import { Link as ReactRouterLink, LinkProps as ReactRouterLinkProps } from 'react-router-dom';

// https://github.com/ReactTraining/react-router/issues/6056
/* istanbul ignore next */
const ReactRouterInternalLink = React.forwardRef<HTMLAnchorElement, ReactRouterLinkProps>((props, ref) => (
  <ReactRouterLink innerRef={ref} {...props} />
));

export interface InternalLinkProps {
  readonly to: string;
}

const InternalLink: React.FC<InternalLinkProps> = ({ to, children }) => (
  <Link component={ReactRouterInternalLink} to={to}>
    {children}
  </Link>
);

export default InternalLink;
