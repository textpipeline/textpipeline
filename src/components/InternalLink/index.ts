export { default } from './InternalLink';
export * from './InternalLink';
