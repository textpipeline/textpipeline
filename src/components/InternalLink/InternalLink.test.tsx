import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import InternalLink from './InternalLink';

describe('InternalLink', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    rendered = shallow(<InternalLink to="/" />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
});
