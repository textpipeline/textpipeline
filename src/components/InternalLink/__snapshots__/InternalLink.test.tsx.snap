// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalLink renders correctly 1`] = `
<WithStyles(ForwardRef(Link))
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  to="/"
/>
`;
